[fix bug 1295181] Test /all links on /firefox/new/.

jpetto opened the pull request:
## Description

Adds two variations to scene 1 of `/firefox/new`. Each variation adds a link underneath the primary download button that:

1. sends the user to `/firefox/all`
2. opens  a modal containing direct download links for all platforms, app store badges, and another primary download button

Understood that the JS is very much a hack right now for the modal variation. If that variation ends up winning, we'll need to figure out a more robust implementation.

Code to serve variations w/out Optimizely will come in a follow-up PR.

## Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1295181

## Testing

The usual. Ensure new links only display when download button is present (i.e. not when using up-to-date Firefox) and only for en-US. Double check old IE.

https://bedrock-demo-jpetto-fxnew.us-west.moz.works/en-US/firefox/new/?v=1
https://bedrock-demo-jpetto-fxnew.us-west.moz.works/en-US/firefox/new/?v=2

## Checklist
- [ ] Related functional & integration tests passing.

jpetto commented:
@alexgibson r?

alexgibson commented on bedrock/firefox/templates/firefox/includes/firefox-footer-links.html:
```diff
@@ -27,7 +27,7 @@
-    <li><a href="{{ firefox_url('desktop', 'all', channel) }}">{{ _('Firefox for Other Platforms & Languages') }}</a></li>
-    <li><a href="{{ firefox_url('desktop', 'notes', channel) }}">{{ _('Release Notes') }}</a></li>
+    <li><a id="fx-footer-links-desktop-all" href="{{ firefox_url('desktop', 'all', channel) }}">{{ _('Firefox for Other Platforms & Languages') }}</a></li>
+    <li><a id="fx-footer-links-release-notes" href="{{ firefox_url('desktop', 'notes', channel) }}">{{ _('Release Notes') }}</a></li>
```
Is this `id` being used?

jpetto replied:
It is not. I added this last minute to be consistent with its sibling, which I'm still not sure is the "right" choice. Happy to remove the unused id.

alexgibson commented on bedrock/firefox/templates/firefox/new/horizon/scene1.html:
```diff
@@ -120,9 +125,42 @@ <h2 class="good">{{_('Freedom is yours') }}</h2>
+        </section>{#-- /#fx-modal-primary-ctas --#}
+
+        <section id="fx-modal-secondary-cta" class="fx-modal-section">
+          <h3>Download Firefox<span id="fx-modal-user-platform"></span></h3>
```
Nit: space here after `Download Firefox`?

jpetto replied:
Will fix. 👍

alexgibson commented on media/css/base/mozilla-modal.less:
```diff
@@ -70,6 +70,7 @@ html.noscroll body {
                     .image-replaced();
                     border-radius: 50%;
                     border: 2px solid #fff;
+                    transition: transform 0.15s ease;
```
Nit: use `.transition()` mixin? It makes no difference on this page, but other's could benefit from it?

jpetto replied:
I'm of the opinion that we deprecate that `.transition` mixin. All major browsers now support the non-prefixed syntax, it's a visual enhancement... What do you think? File a bug to drop the mixin, or should I use it here?

alexgibson replied:
Personally I think Webkit is a bit of a different issue to evergreen desktop browsers. There are so many mobile webkit devices out there that are not updated, it doesn't seem like much hardship to keep the last vendor prefix in there. But I could be persuaded that we're not too concerned about old devices...

alexgibson commented on media/img/firefox/new/horizon/down-arrow.svg:
```diff
@@ -0,0 +1,14 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 100 100">
+  <path
```
You may have done this intentionally, but this svg could be minified to save on whitespece?

jpetto replied:
Ah, good catch. Will compress.

alexgibson commented on media/js/firefox/new/all-test-variations.js:
```diff
@@ -0,0 +1,132 @@
+        var $newLink;
+
+        // make sure desktop download button exists and user is on a recognized platform
+        if ($dlButton.length && $dlButton.find('.unrecognized-download:visible').length === 0) {
```
Should we add a check here to make sure it's the desktop download button that's visible, and not the mobile button?

jpetto replied:
Ah, the CSS for the page hides the desktop download button on iOS and Android, but it wasn't preventing the variation JS from running. Will add a check here to make sure user is not on iOS or Android.

alexgibson commented:
The svg arrows look a little squished in IE:

![arrows](https://cloud.githubusercontent.com/assets/400117/18743333/69229f66-80b0-11e6-8b93-9a5cbe6cc50e.png)

alexgibson commented:
This may be by design, but the link font size is a little smaller than the other privacy link:

![font-size](https://cloud.githubusercontent.com/assets/400117/18743341/74173062-80b0-11e6-87d5-0c02085b95e7.png)

alexgibson commented:
Seems to function as expected 👍 

Code is a little crazy but we knew this experiment would be such at least temporarily. The modal looks a bit broken in IE 6/7 but the platform links within it are still functional so I think that's fine? Other than the review items above, I think this looks good. r+wc 🌵

jpetto commented:
Yeah, font size on the link is as requested. Not too concerned with IE 6 & 7 UI issues (as long as the links work).

jpetto commented:
Marking as Do Not Merge until we verify no other teams need to run tests on this page at the same time.

jpetto commented:
Rebased with master and added Traffic Cop configuration. Would be good to have another once over to make sure all is well before merging (which is slated for Tuesday, 10/18).

alexgibson commented on media/css/firefox/new/all-test-variations.less:
```diff
@@ -0,0 +1,190 @@
+
+        a {
+            .font-size(@largeFontSize);
+            background: url('/media/img/firefox/new/horizon/down-arrow.svg') left center no-repeat;
```
Nit: now that we've removed the old /new template, this asset could just live at `/media/img/firefox/new/`

alexgibson commented on media/js/firefox/new/all-test-variations.js:
```diff
@@ -0,0 +1,133 @@
+        var newOsName;
+
+        switch ($liA.text()) {
+        case 'Windows':
```
This needn't block this PR, but just to note we're adding another download button type in #4413 for sha-1 downloads. I'll likely need to update the logic for what's displayed in this test after this merges to master.

jpetto replied:
Yeah, I figured that new button would impact quite a few things. Hopefully wont be too much work to add it here when the time comes.
